### PR TITLE
feat: add simulate location service

### PR DESCRIPTION
**Pull request by umutuzgur**

### CODE REVIEW

File: lib/simulatelocation/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+//We just need to write any data to the client. It doesn't matter what we send
+const SIMULATE_LOCATION_SERVICE_NAME = 'com.apple.dt.simulatelocation';
+
+const RESET_MESSAGE = Buffer.from([0, 0, 0, 1]);
+const SET_LOCATION_MESSAGE = Buffer.from([0, 0, 0, 0]);
+
+class SimulateLocationService {
```
Comment (mykola-mokhnach) [inline lib/simulatelocation/index.js]: LocationSimulatorService

File: lib/simulatelocation/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  constructor (socketClient) {
+    this._socketClient = socketClient;
+
```
Comment (mykola-mokhnach) [inline lib/simulatelocation/index.js]: extra empty line

File: lib/simulatelocation/index.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    this._socketClient.write(RESET_MESSAGE);
+  }
+
+  /**
+   * Set the mock location on the device
+   * @param {string|number} lat The latitude that wants to be set on the device
+   * @param {string|number} long The longitude that wants to be set on the device
+   */
+  setLocation (lat, long) {
+    lat = lat.toString();
```
Comment (mykola-mokhnach) [inline lib/simulatelocation/index.js]:
```
`${lat}`
```

transformation is safer

File: lib/simulatelocation/index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    long = long.toString();
+
+    const lat_length = Buffer.alloc(4);
```
Comment (mykola-mokhnach) [inline lib/simulatelocation/index.js]: this stuff can actually be done in a loop for both attitude and longitude

Comment (umutuzgur) [reply lib/simulatelocation/index.js]: It can but I think it looks more meaningful like this

File: lib/simulatelocation/index.js (hunk 1 above)
Comment (umutuzgur) [reply lib/simulatelocation/index.js]: Does it make a difference? I mean the service on the phone side is called simulatelocation

**Pull request merged**